apps/src/weblab2/Weblab2View.tsx

```diff
@@ -139,7 +142,7 @@ const Weblab2View: React.FC<
 
   return (
     <div className={moduleStyles.weblab2Container}>
-      {hasSource && (
+      {hasSource && sourceLevel === levelProperties.id && (
```

Comment:
To confirm my understanding...
I repro'd error when going from a sublevel with inactive file: 'demo1', then going to a level with active file 'demo1', then going back to sublevel. When I open 'demo1' in sublevel editor (make active), it has been overwritten.
However, if I go from sublevel to level which has a file named 'demo1' but it is inactive, and then go back to sublevel, the 'demo1' file is NOT overwritten.
This makes sense according to your explanation according to your explanation - only the active file in the sublevel fired an editor update and saved to redux. Thanks so much for digging.
I do have a question ... why does this bug only manifest when going from level to sublevel and not level to level?